data_pipeline/civicrm_email_report/civicrm_email_report.py

```diff
@@ -0,0 +1,74 @@
+import json
+import datetime
+import logging
+from typing import Iterable, Sequence
+import requests
+
+LOGGER = logging.getLogger(__name__)
+
+
+def get_current_timestamp():
+    return datetime.datetime.utcnow().isoformat()
+
+
+def get_mailing_id_dict(mail_id: int) -> str(dict):
+    return json.dumps({'mailing_id': mail_id})
+
+
+def get_connection_parameters(mail_id: int, api_key, site_key):
+    return {
+        "entity": "Mailing",
+        "action": "stats",
+        "json": get_mailing_id_dict(mail_id),
+        "api_key": api_key,
+        "key": site_key
+    }
+
+
+def connect_civiapi_and_get_email_report(
+    url: str,
+    mail_id: int,
+    api_key: str,
+    site_key: str
+) -> dict:
+    response = requests.post(url=url, data=get_connection_parameters(mail_id, api_key, site_key))
+    response.raise_for_status()
+    return response.json()
+
+
+def format_email_report(
```

Comment:
minor: usually I would associate a `format` function with a text output. Perhaps another name could make it more clear? Maybe `transform`? And maybe it could say from what to what format do we transform it?